test/contrib/test_funsor.py

```diff
@@ -0,0 +1,92 @@
+# Copyright Contributors to the Pyro project.
+# SPDX-License-Identifier: Apache-2.0
+
+import numpy as np
+from numpy.testing import assert_allclose
+
+from jax import random
+import jax.numpy as jnp
+
+import numpyro
+from numpyro.contrib.funsor import markov
+import numpyro.distributions as dist
+from numpyro.infer import MCMC, NUTS
+
+
+def test_gaussian_mixture_model():
```

Comment:
Do you consider these tests sufficient for merging this PR?  It's up to you - as I mentioned before, there are a few extra tests in the Pyro version that don't depend on Pyro's old enumeration machinery you could port if you wanted to.

Reply:
Oh, I missed that point. I'll try to port some of them here. Thanks!

Reply:
You can probably just copy this file: https://github.com/pyro-ppl/pyro/pull/2475/files#diff-90ba92d31fd3a057903e4e0663d72b99

Reply:
Thanks, I just port it here.